packaging/tags.py

```diff
@@ -219,16 +247,24 @@ def _mac_binary_formats(version, cpu_arch):
     return formats
 
 
-def _mac_platforms(version=None, arch=None):
+def _mac_platforms(
+    version=None,  # type: Optional[Tuple[int, int]]
+    arch=None,  # type: Optional[str]
+):
+    # type: (...) -> List[str]
     version_str, _, cpu_arch = platform.mac_ver()
     if version is None:
-        version = tuple(map(int, version_str.split(".")[:2]))
+        _version = tuple(map(int, version_str.split(".")[:2]))
```

Comment:
Why is it necessary to use a different variable name throughout the function? It seems like it should be possible to keep using `version` since the type is the same.

Reply:
Because farther down we do `version[0]`, which gives the following error if we retain the original type:

```
Value of type "Optional[Tuple[int, int]]" is not indexable
```

Reply:
You should be able to put `assert version is not None` at an appropriate location, which will let `mypy` resolve `Optional[Tuple[int, int]]` to `Tuple[int, int]`.

Reply:
This feels a bit weird but it does work 👍

Reply:
It’s actually the recommended way! https://github.com/python/typing/issues/645

Reply:
TIL! 😄